### DISCUSSION TITLE
MQTT in Web Admin page

### DISCUSSION THREAD

**Discussion by MichalNemec** [processone/ejabberd#4090] [Questions]
Hello, i just setup ejabberd and it seems like the admin dashboard is pretty much barebones. Is there a way to show connected mqtt clients and their subscriptions, topics, like emqx has?

What is the dashboard used for, since it can just wipe offline messages, create users and see messages, can it be extended to have full blown manager for ejabberd?

**Comment by badlop** [comment]
As you noticed, ejabberd's WebAdmin is useful to view the general state of the server and perform a few basic tasks.

But to perform most administrative tasks, there is the command API (see [API Tags](https://docs.ejabberd.im/developer/ejabberd-api/admin-tags/)). Those commands can be executed using the  `ejabberdctl` script, ReST calls, XMP-RPC, and WebAdmin (thanks to [mod_shcommands](https://github.com/processone/ejabberd-contrib/tree/master/mod_shcommands#readme)).

Obviously, using commands for one or two administrative tasks is less friendly than a web. But if they are tasks that should be repeated or automatized, they are preferable to a web page that only a human can use.

Regarding the MQTT module, it has no administrative frontend at all, neither API commands or WebAdmin pages. Lacking any page to view statistics, and commands to perform admin tasks could be a feature request.

On the other hand, to view raw details, if MQTT is stored in the internal mnesia database, the tables numbers can be seen in WebAdmin -> Nodes -> your node -> Database.

**Comment by MichalNemec** [comment]
Thank you for this extensive explanation. The nodes database really helped, because now i can with mnesia verify that my stuff is working as intended. Now i understand the direction ejabberd took. Its up to the developers to create their own "admin panel" exactly how they need with data they need to see (grafana, etc..)
